src/components/user.js

```diff
@@ -15,69 +26,143 @@ export default class User extends React.Component {
         const { info = {}, address = {}, company = {} } = this.props;
 
         this.state = {
-            currentUser: {
+            currentUser:  {
                 info,
                 address,
                 company
             },
-            showAddress: false,
-            showCompany: false,
-            wantEdit:    false
+            formErrors:   {
+                name:          '',
+                email:         '',
+                phone:         '',
+                duplicateUser: {
+                    text:   '',
+                    userId: ''
+                }
+            },
+            showAddress:  false,
+            showCompany:  false,
+            wantEdit:     false,
+            addDuplicate: false
         };
 
-        this.handleClickAddress = this.handleClickAddress.bind(this);
-        this.handleClickCompany = this.handleClickCompany.bind(this);
-        this.handleChange       = this.handleChange.bind(this);
-        this.handleClickEdit    = this.handleClickEdit.bind(this);
-        this.handleClickSubmit  = this.handleClickSubmit.bind(this);
+        this._handleClickAddress = this._handleClickAddress.bind(this);
+        this._handleClickCompany = this._handleClickCompany.bind(this);
```

Comment:
What about getting rid of `binding`

> You use arrow functions in user-list component instead of bindings (see _onChange method)
> Let's use here too